tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i

```diff
@@ -217,6 +211,8 @@
   steady_state_detection = true
   dt = 0.25
   solve_type = PJFNK
+  automatic_scaling = true
+  compute_scaling_once = false
```

Comment:
So we saw that this made the solve significaintly better. For further debugging I would be curious how this does with `-pc_type lu`. For those time steps that we saw taking a lot of linear iterations, I would be curious to know how Jacobian-free is doing at that point. I'm wondering whether at that point whether our Jacobian-free approximation is doing a poor job of representing the Jacobian. It would be good to run with `Outputs/print_linear_residuals=false -ksp_monitor_true_residual` and see whether the true residual is dropping. Does this problem get more and more non-linear over time?

Reply:
> Jacobian-free approximation is doing a poor job of representing the Jacobian

Sorry this would show up more in the non-linear iterations, not the linears. But you can see cases where the GMRES residuals drop but the true residuals do not; this usually indicates that your Jacobian-free process is non-linear, e.g. due to finite differencing of a noisy function or something. With good scaling though, that should be less of a concern.

Actually the most obvious explanation for a degrading linear solve would be if our off-diagonal terms are getting stronger, since we are currently only using the diagonals in the preconditioning matrix. Is that possibly happening? Won't happen in time for training, but we definitely want to figure out where our Jacobian bug is! Then we could run with NEWTON :smile:

Reply:
Thanks for digging into this, I think what I have in this PR is good for the training. We will get to this example sometime on Tue., so you might be able to help with this section and could talk about solver options and such.

Reply:
-pc_type lu works much better, it doesn't seem to get worse over time.